netbox/dcim/forms.py

```diff
@@ -2843,6 +3006,19 @@ def __init__(self, *args, **kwargs):
         self.fields['untagged_vlan'].widget.add_additional_query_param('site_id', device.site.pk)
         self.fields['tagged_vlans'].widget.add_additional_query_param('site_id', device.site.pk)
 
+    def clean(self):
+
+        # Validate that the number of ports being created from both the name_pattern and label_pattern are equal
+        name_pattern_count = len(self.cleaned_data['name_pattern'])
+        label_pattern_count = len(self.cleaned_data['label_pattern'])
+        if label_pattern_count and name_pattern_count != label_pattern_count:
+            raise forms.ValidationError({
+                'label_pattern': 'The provided name pattern will create {} interfaces, however {} labels will '
+                'be generated. These counts must match.'.format(
+                    name_pattern_count, label_pattern_count)
+            })
+
+
```

Comment:
Since we are repeating the almost exact same code block (with one word different int he exception), it perhaps might be better to have a common class for this.  Or at least to merge the common ones.

Reply:
I purposefully avoided modifying existing code when possible. Do you recommend creating a new form Class that all these would inherit from ?

Reply:
@DanSheps https://github.com/netbox-community/netbox/pull/4723/commits/4301c06d177439356d4f4da207f2af1128984293 should make you happy :)

Reply:
We may want to add [field_order](https://docs.djangoproject.com/en/3.0/ref/forms/api/#django.forms.Form.field_order) to the forms though as this shifts the usual first field after `name` and `label`. Let me know.